### PR TITLE
BrainFPV RE1 Support (V3)

### PR DESCRIPTION
**Pull request by mluessi**
Adds support for the BrainFPV RE1 (Racing Edition 1) flight controller. RE1 is an evolution of Brain and has also has a built-in OSD with a lot of other enhancements. Some details: STM32F446 running at 180MHz, gyro / accel: Bosch BMI160 running at 1.6kHz, 128Mbit of flash for logging etc. RE1 also contains additional custom hardware that allows it to control RGB LEDs (WS2812B) over SPI and also an IR emitter ("transponder") for lap timing when racing. The LEDs and the emitter can be controlled using the OSD menu.

Supersedes #965 rebased on `next` and added battery defaults as well as support for BMI160 fast offset compensation (will be performed by BrainFPV).

### CODE REVIEW

**Comment by mlyle** [comment]
jenkins, build artifacts please

File: flight/targets/brainre1/fw/pios_board.c [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+ */
+static void settingdUpdatedCb(UAVObjEvent * ev, void *ctx, void *obj, int len)
+{
+	(void) ev; (void) ctx; (void) len;
+
+	HwBrainRE1Data * hwre1data = (HwBrainRE1Data *)obj;
+
+	/* Set the colors of the RGB LEDs */
+	if (hwre1data->NumberOfLEDs > 0) {
+		uint8_t led_color[3];
```
Comment (mlyle) [inline flight/targets/brainre1/fw/pios_board.c]: For future.. i prefer `= {};` or `= {0};` to memset below.  Easier for compiler to infer/elide.  (not blocker for this).

File: flight/targets/brainre1/target-defs.mk [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Add this board to the list of buildable boards
+ALL_BOARDS += brainre1
+
+# Set the cpu architecture here that matches your STM32
+# Should be one of: f1,f3,f4
+brainre1_cpuarch := f4
+
+# Short name of this board (used to display board name in parallel builds)
+# Should be exactly 4 characters long.
+brainre1_short := bre1
```
Comment (mlyle) [inline flight/targets/brainre1/target-defs.mk]: need newline at end of file.

File: flight/targets/brainre1/re1fpga/fpga_drv.c [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+
+#include <pios.h>
+
+#if defined(PIOS_INCLUDE_RE1_FPGA)
+
+#include "fpga_drv.h"
+#include "stm32f4xx_rcc.h"
+#include "pios_spi_priv.h"
+
+/**
```
Comment (mlyle) [inline flight/targets/brainre1/re1fpga/fpga_drv.c:42]: Thank you for this register map; very useful/necessary IMO.

**Comment by mlyle** [comment]
Only blocker I see is the newline issue.  Ideally someone would fly this and say it is OK, then I say "merge it".

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
Jenkins is also broken, which is a blocker now :P

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by fujin** [comment]
Tested asset on win10 x64 :ok: 

image

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
test this please

**Comment by mlyle** [comment]
test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
Super sorry for all the spam, everyone.  It seems that jenkins ghprb changed the case sensitivty behavior.  We couldn't downgrade the plugin either because there are (undocumented) version dependency issues.

**Comment by mlyle** [comment]
Congrats-- and thanks for your patience with getting this merged.  I'm not 100% comfortable yet, but we have some time to make sure everythign is right.

**Pull request merged**